patches/server/0978-add-getHighestMapID-method.patch

```diff
@@ -0,0 +1,45 @@
+From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
+From: chmodsayshello <chmodsayshello@hotmail.com>
+Date: Mon, 12 Jun 2023 18:56:48 +0200
+Subject: [PATCH] add getHighestMapID method
+
+
+diff --git a/src/main/java/net/minecraft/world/level/saveddata/maps/MapIndex.java b/src/main/java/net/minecraft/world/level/saveddata/maps/MapIndex.java
+index 9b2948b5150c8f039ca667a50765109721b93947..9f69db46239d3e8055d5793a6a9ba62d321b5e4f 100644
+--- a/src/main/java/net/minecraft/world/level/saveddata/maps/MapIndex.java
++++ b/src/main/java/net/minecraft/world/level/saveddata/maps/MapIndex.java
+@@ -40,4 +40,10 @@ public class MapIndex extends SavedData {
+         this.setDirty();
+         return i;
+     }
++
++    // Paper start - get value without writing
++    public int getCurrentValueForMap() {
++        return this.usedAuxIds.getInt("map");
++    }
++    // Paper end
+ }
+diff --git a/src/main/java/org/bukkit/craftbukkit/CraftServer.java b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+index ffa27c9c02dc4d12411fc089de3af8e8e12ba06e..790271eaad20c8276fcae597b5ff79a81eaf0dfb 100644
+--- a/src/main/java/org/bukkit/craftbukkit/CraftServer.java
++++ b/src/main/java/org/bukkit/craftbukkit/CraftServer.java
+@@ -105,6 +105,7 @@ import net.minecraft.world.level.levelgen.WorldDimensions;
+ import net.minecraft.world.level.levelgen.WorldOptions;
+ import net.minecraft.world.level.material.Fluid;
+ import net.minecraft.world.level.saveddata.maps.MapDecoration;
++import net.minecraft.world.level.saveddata.maps.MapIndex;
+ import net.minecraft.world.level.saveddata.maps.MapItemSavedData;
+ import net.minecraft.world.level.storage.LevelStorageSource;
+ import net.minecraft.world.level.storage.PlayerDataStorage;
+@@ -2971,5 +2972,11 @@ public final class CraftServer implements Server {
+         return this.potionBrewer;
+     }
+ 
++    @Override
++    public int getHighestMapID(){
++        // Paper - all map IDs are meant for the overworld, this line is more or less copied from NMS
++        // Paper - for additional info, view getFreeMapId() in ServerLevel.java and getFreeAuxValueForMap() in MapIndex.java
++        return ((MapIndex) this.getServer().overworld().getDataStorage().computeIfAbsent(MapIndex::load, MapIndex::new, "idcounts")).getCurrentValueForMap();
++    }
```

Comment:
Properly surround this in paper comments

Reply:
I put this in an already existing paper block... Thats why here, it only shows paper end.

In my opinion doing something like:
```
// Paper End
// Paper Start
<code>
// Paper End
```
is silly! Now, should I still do so?

Reply:
That's fine then